test/blackbox-tests/test-cases/shared-object-package/run.t

```diff
@@ -0,0 +1,3 @@
+  $ jbuilder build
```

Comment:
Can you add a comment with the expected behavior?

Reply:
nvm, I thought this PR was just to add the test case, I missed the WIP. I suppose this will disappear once it's finished

Reply:
Yes, absolutely. I noticed several weird behaviors when trying to install a DLL, but the root cause seems that dune does not know how to do that, so I guess that PR will be about this.